google/resource_compute_region_instance_group_manager.go

```diff
@@ -0,0 +1,483 @@
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"port": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"self_link": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"target_pools": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Set: selfLinkRelativePathHash,
+			},
+
+			"target_size": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+
+			"auto_healing_policies": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"health_check": &schema.Schema{
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: compareSelfLinkRelativePaths,
+						},
+
+						"initial_delay_sec": &schema.Schema{
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(0, 3600),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceComputeRegionInstanceGroupManagerCreate(d *schema.ResourceData, meta interface{}) error {
+	computeApiVersion := getComputeApiVersion(d, RegionInstanceGroupManagerBaseApiVersion, RegionInstanceGroupManagerVersionedFeatures)
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	manager := &computeBeta.InstanceGroupManager{
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		BaseInstanceName:    d.Get("base_instance_name").(string),
+		InstanceTemplate:    d.Get("instance_template").(string),
+		TargetSize:          int64(d.Get("target_size").(int)),
+		NamedPorts:          getNamedPortsBeta(d.Get("named_port").([]interface{})),
+		TargetPools:         convertStringSet(d.Get("target_pools").(*schema.Set)),
+		AutoHealingPolicies: expandAutoHealingPolicies(d.Get("auto_healing_policies").([]interface{})),
+		// Force send TargetSize to allow size of 0.
+		ForceSendFields: []string{"TargetSize"},
+	}
+
+	var op interface{}
+	switch computeApiVersion {
+	case v1:
+		managerV1 := &compute.InstanceGroupManager{}
+		err = Convert(manager, managerV1)
+		if err != nil {
+			return err
+		}
+		managerV1.ForceSendFields = manager.ForceSendFields
+		op, err = config.clientCompute.RegionInstanceGroupManagers.Insert(project, d.Get("region").(string), managerV1).Do()
+	case v0beta:
+		op, err = config.clientComputeBeta.RegionInstanceGroupManagers.Insert(project, d.Get("region").(string), manager).Do()
+	}
+
+	if err != nil {
+		return fmt.Errorf("Error creating RegionInstanceGroupManager: %s", err)
+	}
+
+	d.SetId(manager.Name)
```

Comment:
Couldn't we hypothetically have two IGMs in different regions with the same name? What if we use the "canonical link" that @selmanj used for #378?

Reply:
Name is unique globally, I was the one who actually wrote #378. 

I am working on a RegionalId and ZonalId class to standardize all of this. I left it to simply the name for now to mimic the behavior of the single zone resource (`resource_compute_instance_group_manager`). Once I am done with the reusable id classes, I will change all our resources to have consistent and more user-friendly import behavior.

Reply:
> Name is unique globally.

Then ignore me entirely. :D 

> FYI: I was the one who actually wrote #378 ;)

Well, this is mortifying. Apologies. :( 

> I am working on a RegionalId and ZonalId class to standardize all of this. I left it to simply the name for now to mimic the behavior of the single zone resource (resource_compute_instance_group_manager). Once I am done with the reusable id classes, I will change all our resources to have consistent and more user-friendly import behavior.

Sounds good to me!